quickstart/src/main/java/com/microsoft/azure/kusto/quickstart/KustoSampleApp.java

```diff
@@ -0,0 +1,395 @@
+                            System.out.printf("Failed to create a %s mapping reference named '%s'. Skipping this ingestion.%n", ingestionMappingKind, mappingName);
+                            continue;
+                        }
+                    }
+                } else if (StringUtils.isBlank(mappingName)) {
+                    System.out.println("The configuration indicates an existing mapping should be used, but none was provided. Skipping this ingestion.");
+                    continue;
+                }
+
+                // Learn More: For more information about ingesting data to Kusto in Java, see: https://docs.microsoft.com/azure/data-explorer/java-ingest-data
+                String sourceType = file.get("sourceType");
+                String uri = file.get("dataSourceUri");
+                waitForUserToProceed(String.format("Ingest '%s' from %s", uri, sourceType));
+                // Tip: When ingesting json files, if each line represents a single-line json, use MULTIJSON format even if the file only contains one line.
+                //  If the json contains whitespace formatting, use SINGLEJSON. In this case, only one data row json object is allowed per file.
+                if (dataFormat == IngestionProperties.DataFormat.json) {
+                    dataFormat = IngestionProperties.DataFormat.multijson;
+                }
+
+                // Tip: Avoid creating a new ingest Client for each use. Instead, create the client once and use it as long as possible.
+                if (sourceType.equalsIgnoreCase("localfilesource")) {
+                    ingestDataFromFile(ingestClient, databaseName, tableName, uri, dataFormat, mappingName);
+                } else if (sourceType.equalsIgnoreCase("blobsource")) {
+                    ingestDataFromBlob(ingestClient, databaseName, tableName, uri, dataFormat, mappingName);
+                } else {
+                    System.out.printf("Unknown source '%s' for file '%s'%n", sourceType, uri);
+                }
+            }
+        } catch (URISyntaxException | IOException e) {
+            die("Couldn't create Kusto client", e);
+        }
+
+        System.out.printf("Sleeping %s seconds for queued ingestion to complete. Note: This may take longer depending on the file size and ingestion policy.%n", WAIT_FOR_INGEST_SECONDS);
+        for (int i = WAIT_FOR_INGEST_SECONDS; i >= 0; i--) {
+            System.out.printf("%s.", i);
+            try {
+                Thread.sleep(TimeUnit.SECONDS.toMillis(1));
+            } catch (InterruptedException e) {
+                System.out.println("Exception while sleeping waiting for ingestion, with stack trace:");
+                e.printStackTrace();
+                throw e;
+            }
+        }
+        System.out.println();
+
+        System.out.printf("Step %s: Get post-ingestion row count for '%s.%s'%n", step++, databaseName, tableName);

```

Comment:
there were like a million steps that occurred before this ++ :)

Reply:
Each of them is incremented via `waitForUserToProceed()`